src/App.tsx

```diff
@@ -9,39 +9,44 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
+
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from './app/store';
```

Comment:
Ensure that `AppDispatch` and `RootState` are correctly defined in the `store` file.